tuf-cli/get.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/flynn/go-docopt"
+	tuf "github.com/flynn/go-tuf/client"
+	"github.com/flynn/go-tuf/util"
+)
+
+func init() {
+	register("get", cmdGet, `
+usage: tuf-cli get [-s|--store=<path>] <url> <target>
+
+Options:
+  -s <path>    The path to the local file store [default: tuf.db]
+
+Get a target from the repository.
+  `)
+}
+
+type tmpFile struct {
+	*os.File
+}
+
+func (t *tmpFile) Delete() error {
+	t.Close()
+	return os.Remove(t.Name())
+}
+
+func cmdGet(args *docopt.Args, client *tuf.Client) error {
+	if _, err := client.Update(); err != nil && !tuf.IsLatestSnapshot(err) {
+		return err
+	}
+	target := util.NormalizeTarget(args.String["<target>"])
+	tmp, err := ioutil.TempFile("", "go-tuf")
+	if err != nil {
+		return err
+	}
+	dest := tmpFile{tmp}
+	if err := client.Download(target, &dest); err != nil {
+		return err
+	}
+	defer tmp.Close()
+	if _, err := tmp.Seek(0, os.SEEK_SET); err != nil {
+		return err
+	}
+	_, err = io.Copy(os.Stdout, tmp)
```

Comment:
I usually prefer commands to be filters by default. It can always be piped to a file on the command line if needed, and it is less code. WDYT?